src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -55,122 +61,172 @@ public static SyncerFactory of(
+          .map(
+              syncOption -> {
+                if (isBlank(syncOption)) {
+                  final String errorMessage =
+                      format(
+                          "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
 
-      return exceptionallyCompletedFuture(new IllegalArgumentException(errorMessage));
+                  throw new IllegalArgumentException(errorMessage);
+                }
+
+                final String trimmedValue = syncOption.trim();
+                try {
+                  return SyncModuleOption.getSyncModuleOptionBySyncOptionValue(trimmedValue);
+                } catch (IllegalArgumentException e) {
+                  final String errorMessage =
+                      format(
+                          "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+                          trimmedValue,
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
+                  throw new IllegalArgumentException(errorMessage);
+                }
+              })
+          .collect(Collectors.toList());
     }
+  }
 
-    Syncer<
-            ? extends ResourceView,
-            ?,
-            ? extends BaseSyncStatistics,
-            ? extends BaseSyncOptions<?, ?>,
-            ? extends QueryDsl<?, ?>,
-            ? extends BaseSync<?, ?, ?>>
-        syncer;
+  private static Collection<List<SyncModuleOption>> groupSyncModuleOptions(
+      List<SyncModuleOption> syncModuleOptions) {
+    Map<SyncModuleOption, Integer> syncModuleOptionWithReferencesCount =
+        syncModuleOptions
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    key -> key,
+                    option ->
+                        (option.getReferences().isEmpty())
+                            ? 0
+                            : countReferences(option.getReferences(), syncModuleOptions)));
 
-    try {
-      syncer = buildSyncer(syncOptionValue, runnerNameOptionValue, isSyncProjectSyncCustomObjects);
-    } catch (IllegalArgumentException exception) {
+    Map<Integer, List<SyncModuleOption>> groupedSyncModuleOptions = new TreeMap<>();
+    for (Map.Entry<SyncModuleOption, Integer> syncOptionWithReferenceCount :
+        syncModuleOptionWithReferencesCount.entrySet()) {
+      if (!groupedSyncModuleOptions.containsKey(syncOptionWithReferenceCount.getValue())) {
+        List<SyncModuleOption> syncOptionsList = new ArrayList<>();
+        for (Map.Entry<SyncModuleOption, Integer> syncModuleOptionIntegerEntry :
+            syncModuleOptionWithReferencesCount.entrySet()) {
+          if (0
+              == syncOptionWithReferenceCount
```

Comment:
`syncOptionWithReferenceCount.getValue()` is an Integer. 
`syncModuleOptionIntegerEntry.getValue()` is also an integer. Here, we are checking whether those 2 integers are equal. isn't it? If so, why not use `equals()` method? 

`if (syncOptionWithReferenceCount.getValue().equals(syncModuleOptionIntegerEntry.getValue())) {`

Reply:
Done https://github.com/commercetools/commercetools-project-sync/pull/219/commits/c8cf9b1275ce81a377a5518785bf1b5f3e7f3fd0